source/slang/slang-check-conversion.cpp

```diff
@@ -687,14 +746,32 @@ namespace Slang
            !canCoerce(toType, fromInitializerListExpr->type, nullptr))
             return _failedCoercion(toType, outToExpr, fromInitializerListExpr);
 
-        if(!_readAggregateValueFromInitializerList(toType, outToExpr, fromInitializerListExpr, argIndex))
+        if(!_readAggregateValueFromInitializerList<false>(toType, outToExpr, fromInitializerListExpr, argIndex))
```

Comment:
should we aalso try legacy logic here too?

Reply:
we run modern logic here.

below we run legacy logic, we run legacy logic if modern logic fails

Reply:
but isn't this saying if modern logic failed, return false?

Reply:
it seems what the code is doing is that if modern logic succeeds but there are still remaining args left, we used to throw an error, now we try legacy.

Reply:
I don't know if we need to handle the other case (return false) for when toType is a `StructType`?

Reply:
I am confused. when `_readAggregateValueFromInitializerList` returns false, it means the modern approach failed.
Then shoudln't we try legacy logic now, instead of returning `false`?

Reply:
`_readAggregateValueFromInitializerList<true>` is only invoked when `_readAggregateValueFromInitializerList<false>` returns `true` and `argIndex != argCount`? Is this correct?

Reply:
I understand your point, will make a new commit with the addition

Reply:
already fixed.